src/lib/sanity/queries/fragments/fragment.types.ts

```diff
@@ -0,0 +1,30 @@
+import { GetPageQueryResult, PostQueryResult, PostsArchiveQueryResult } from '@/sanity.types';
+
+export type PostCardFragmentType = NonNullable<PostsArchiveQueryResult['results'][number]>;
+export type PostFragmentType = NonNullable<PostQueryResult>;
+export type PersonFragmentType = NonNullable<PostFragmentType['author']>;
+export type CategoryFragmentType = NonNullable<PostFragmentType['categories']>[number];
+
+export type PageFragmentType = NonNullable<GetPageQueryResult>;
```

Comment:
I'm not sure if this one makes sense. if the result of a query can be null, we would want the code to check for it.

Reply:
If you need this just for the following types, make don't make this one exportable.

Reply:
@nicholasio 
> if the result of a query can be null, we would want the code to check for it.

Yes we definitely want to check for null. The types returned from `sanityFetch()` would still return a union type that includes null.
the use case for this type could be a `<PageTemplate pageData={pageData satisfies PageFragmentType} />` component. To simplify the `<PageTemplate>` component, the null check would not happen within that page template but in the wrapping `page.tsx`